src/rules/noUnnecessaryElseRule.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2018 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        description: Lint.Utils.dedent`
+            Disallows else block when the If block contains control flow statements, such as \`return\`, \`continue\`,
+            \`break\` or \`throw\`.`,
+        descriptionDetails: "",
+        optionExamples: [true],
+        options: null,
+        optionsDescription: "Not configurable.",
+        rationale: Lint.Utils.dedent`
+            When control flow statements,
+            such as \`return\`, \`continue\`, \`break\` or \`throw\` is written inside \`if\` block,
+            Then \`else\` block becomes unnecessary.`,
+        ruleName: "no-unnecessary-else",
+        type: "functionality",
+        typescriptOnly: false,
+    };
+    /* tslint:disable:object-literal-sort-keys */
+
+    public static FAILURE_STRING(name: string): string {
+        return `If block contains \`${name}\` statement. Consider replacing the contents of else block outside of the block.`;
```

Comment:
The rule doesn't need to say "consider" because the rule failure is a definite "you must do this":

```suggestion
        return `The preceding \`if\` block ends with a \`${name}\` statement. This \`else\` block is unnecessary.`;
```

Reply:
Done https://github.com/palantir/tslint/pull/4502/commits/a1ec00c3360cd8d8b7e36e4ea7b18410a14db156